src/Microsoft.TestPlatform.Build/Tasks/TestTaskUtils.cs

```diff
@@ -0,0 +1,222 @@
+                        dumpArgs.Add($"HangDumpType={task.VSTestBlameHangDumpType}");
+                    }
+
+                    if (!task.VSTestBlameHangTimeout.IsNullOrEmpty())
+                    {
+                        dumpArgs.Add($"TestTimeout={task.VSTestBlameHangTimeout}");
+                    }
+                }
+            }
+
+            if (dumpArgs.Any())
+            {
+                builder.AppendSwitchIfNotNull("--Blame:", string.Join(";", dumpArgs));
+            }
+            else
+            {
+                builder.AppendSwitch("--Blame");
+            }
+        }
+
+        if (task.VSTestCollect != null && task.VSTestCollect.Any())
+        {
+            foreach (var arg in task.VSTestCollect)
+            {
+                if (arg != null)
+                {
+                    // For collecting code coverage, argument value can be either "Code Coverage" or "Code Coverage;a=b;c=d".
+                    // Split the argument with ';' and compare first token value.
+                    var tokens = arg.Split(';');
+
+                    if (arg.Equals(codeCoverageString, StringComparison.OrdinalIgnoreCase) ||
+                        tokens[0].Equals(codeCoverageString, StringComparison.OrdinalIgnoreCase))
+                    {
+                        isCollectCodeCoverageEnabled = true;
+                    }
+                }
+                builder.AppendSwitchIfNotNull("--collect:", arg);
+            }
+        }
+
+        if (isCollectCodeCoverageEnabled || isRunSettingsEnabled)
+        {
+            // Pass TraceDataCollector path to vstest.console as TestAdapterPath if --collect "Code Coverage"
+            // or --settings (User can enable code coverage from runsettings) option given.
+            // Not parsing the runsettings for two reason:
+            //    1. To keep no knowledge of runsettings structure in VSTestTask.
+            //    2. Impact of adding adapter path always is minimal. (worst case: loads additional data collector assembly in datacollector process.)
+            // This is required due to currently trace datacollector not ships with dotnet sdk, can be remove once we have
+            // go code coverage x-plat.
+            if (task.VSTestTraceDataCollectorDirectoryPath != null && !task.VSTestTraceDataCollectorDirectoryPath.ItemSpec.IsNullOrEmpty())
+            {
+                builder.AppendSwitchIfNotNull("--testAdapterPath:", task.VSTestTraceDataCollectorDirectoryPath);
+            }
+            else
+            {
+                if (isCollectCodeCoverageEnabled)
+                {
+                    // Not showing message in runsettings scenario, because we are not sure that code coverage is enabled.
+                    // User might be using older Microsoft.NET.Test.Sdk which don't have CodeCoverage infra.
+                    task.Log.LogWarning(Resources.Resources.UpdateTestSdkForCollectingCodeCoverage);
+                }
+            }
```

Comment:
If could be merged with outer else

```suggestion
            else if (isCollectCodeCoverageEnabled)
            {
                // Not showing message in runsettings scenario, because we are not sure that code coverage is enabled.
                // User might be using older Microsoft.NET.Test.Sdk which don't have CodeCoverage infra.
                task.Log.LogWarning(Resources.Resources.UpdateTestSdkForCollectingCodeCoverage);
            }
```